### ISSUE TITLE
RuntimeError: expected scalar type Float but found Half

### ISSUE THREAD

**Issue by JKHeadley** [lllyasviel/Fooocus#434]
Just ran "run.bat" and then used "a cat" for a prompt and clicked generate.

Below are my terminal logs. Please let me know if you need any more info.

```
C:\Users\headl\Documents\AI\Fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Already up-to-date
Update succeeded.
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.0.65
Inference Engine exists.
Inference Engine checkout finished.
Detected GPU Type: NVIDIA GPU
Total VRAM 4096 MB, total RAM 16086 MB
Trying to enable lowvram mode because your GPU seems to have 4GB or less. If you don't want this use: --normalvram
xformers version: 0.0.20
Set vram state to: LOW_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3050 Laptop GPU : native
Using xformers cross attention
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
[Virtual Memory System] Logic target is CPU, memory = 16085.515625
[Virtual Memory System] Activated = True
Fooocus Text Processing Pipelines are retargeted to cuda:0
model_type EPS
adm 2560
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
missing {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids'}
Refiner model loaded: C:\Users\headl\Documents\AI\Fooocus\Fooocus\models\checkpoints\sd_xl_refiner_1.0_0.9vae.safetensors
[Virtual Memory System] time = 1.08921s: model released from cpu: C:\Users\headl\Documents\AI\Fooocus\Fooocus\models\checkpoints\sd_xl_refiner_1.0_0.9vae.safetensors
[Virtual Memory System] time = 0.31133s: refiner_clip released from cpu: C:\Users\headl\Documents\AI\Fooocus\Fooocus\models\checkpoints\sd_xl_refiner_1.0_0.9vae.safetensors
model_type EPS
adm 2816
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
missing {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids'}
Base model loaded: C:\Users\headl\Documents\AI\Fooocus\Fooocus\models\checkpoints\sd_xl_base_1.0_0.9vae.safetensors
LoRAs loaded: [('sd_xl_offset_example-lora_1.0.safetensors', 0.5), ('None', 0.5), ('None', 0.5), ('None', 0.5), ('None', 0.5)]
Fooocus Expansion engine loaded for cuda:0.
App started successful. Use the app with http://127.0.0.1:7860/ or 127.0.0.1:7860
args:  ('a cat', '', ['Fooocus V2', 'Default (Slightly Cinematic)'], 'Speed', '1152×896', 2, 266721999, 2, 'sd_xl_base_1.0_0.9vae.safetensors', 'sd_xl_refiner_1.0_0.9vae.safetensors', 'sd_xl_offset_example-lora_1.0.safetensors', 0.5, 'None', 0.5, 'None', 0.5, 'None', 0.5, 'None', 0.5, False, 'uov', 'Disabled', None, [], None)
[Fooocus] Initializing ...
[Fooocus] Loading models ...
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] New suffix: intricate, highly detailed, digital painting, artstation, concept art, smooth, sharp focus, illustration, unreal engine 5, 8 k, art by artgerm and greg rutkowski and alphonse mucha
[Fooocus] Preparing Fooocus text #2 ...
[Prompt Expansion] New suffix: extremely detailed digital painting, in the style of fenghua zhong and ruan jia and jeremy lipking and peter mohrbacher, mystical colors, rim light, beautiful lighting, 8 k, stunning scene, raytracing, octane, trending on artstation
[Fooocus] Encoding base positive #1 ...
[Fooocus] Encoding base positive #2 ...
[Fooocus] Encoding base negative #1 ...
[Fooocus] Encoding base negative #2 ...
[Virtual Memory System] time = 0.52184s: refiner_clip loaded to cpu: C:\Users\headl\Documents\AI\Fooocus\Fooocus\models\checkpoints\sd_xl_refiner_1.0_0.9vae.safetensors
[Fooocus] Encoding refiner positive #1 ...
Exception in thread Thread-2 (worker):
Traceback (most recent call last):
  File "threading.py", line 1016, in _bootstrap_inner
  File "threading.py", line 953, in run
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\modules\async_worker.py", line 383, in worker
    handler(task)
  File "C:\Users\headl\Documents\AI\Fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\headl\Documents\AI\Fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\modules\async_worker.py", line 311, in handler
    t['c'][1] = pipeline.clip_encode(sd=pipeline.xl_refiner, texts=t['positive'],
  File "C:\Users\headl\Documents\AI\Fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\headl\Documents\AI\Fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\modules\default_pipeline.py", line 152, in clip_encode
    cond, pooled = clip_encode_single(clip, text)
  File "C:\Users\headl\Documents\AI\Fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\headl\Documents\AI\Fooocus\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\modules\default_pipeline.py", line 128, in clip_encode_single
    result = clip.encode_from_tokens(tokens, return_pooled=True)
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\sd.py", line 556, in encode_from_tokens
    cond, pooled = self.cond_stage_model.encode_token_weights(tokens)
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\sdxl_clip.py", line 85, in encode_token_weights
    g_out, g_pooled = self.clip_g.encode_token_weights(token_weight_pairs_g)
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\modules\patch.py", line 212, in encode_token_weights_patched_with_a1111_method
    out, pooled = self.encode(to_encode)
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\sd1_clip.py", line 161, in encode
    return self(tokens)
  File "C:\Users\headl\Documents\AI\Fooocus\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\headl\Documents\AI\Fooocus\Fooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\sd1_clip.py", line 157, in forward
    pooled_output = pooled_output.to(self.text_projection.device) @ self.text_projection
RuntimeError: expected scalar type Float but found Half
```

**Comment by lllyasviel** [comment]
fixed in 2.0.68

**Comment by JKHeadley** [comment]
It works! Fantastic! Amazing project. Thanks so much!
